_posts/2016-04-07-developing-nectar-extensions-part-1.md

```diff
@@ -0,0 +1,170 @@
+1. _[Extension Framework Game Plan](http://vinsol.com/blog/2016/04/12/extension-framework-game-plan/)_
+1. _[Introduction to Metaprogramming](http://vinsol.com/blog/2016/04/14/introduction-to-metaprogramming/)_
+1. _[Ecto Model Schema Extension](http://vinsol.com/blog/2016/04/15/ecto-model-schema-extension/)_
+1. Ecto Model Support Functions Extension
+1. Phoenix Router Extension
+1. Phoenix View Extension
+1. Running Multiple Elixir Apps Together
+1. Extension Approach Explained
+1. **Developer Experience and Workflow developing Favorite Product Extension**
+1. Developer Experience and Workflow testing Favorite Product Extension
+
+
+
+Developing NectarCommerce Extensions Part 1
+=================
+
+
+### Where we left off ###
+
+
+In the past few blogs we have learned how to write code that extends existing models, routers, added methods to override view rendering and run multiple phoenix application together in the same umbrella project. Let's Continue to build upon that and write our first extension for NectarCommerce [favorite products](http://vinsol.com/blog/2016/04/12/extension-framework-game-plan/) and ultimately our store based on NectarCommerce.
+
+
+### A layered guide to NectarCommerce extensions ###
+
+
+__Setup__: Create a new phoenix application to hold the favorite products application, in your shell run inside the umbrella/apps folder:
+
+<script src="https://gist.github.com/nimish-mehta/994e51defad0787eb88e6611219066fb.js?file=new_phoenix_application.bash"></script>
+
+We could have gone with a regular mix application, but Phoenix/Ecto will come in handy in this case, since we want to have views to display stuff and a model to store data.
+
+While we are at it let's configure our dev.exs and test.exs to use the same db as Nectar, we could write some code and share the db settings between Nectar and our extensions see: [running multiple phoenix application together]() for more details. But now for simplicity's sake we are  just copying the settings from Nectar to get started.
+
+__DB_SETTINGS__:
+
+<script src="https://gist.github.com/nimish-mehta/49dcc6c0bcf6123f536ccc13220bf7ea.js"></script>
+
+We need to let the extension manager know that this application is an extension for Nectar.
+Update the dependencies in extension\_manager/mix.exs with the favorite_products dependency.
+
+<script src="https://gist.github.com/nimish-mehta/418685331be5beb327c2890bc2257b0f.js"></script>
+
+That should be enough to get us going.
+
+__MODEL LAYER__: We want a Nectar user to have some products to like and a way to remember them in short a join table and with two associations let's generate them:
+
+<script src="https://gist.github.com/nimish-mehta/994e51defad0787eb88e6611219066fb.js?file=model_gen.bash"></script>
+
+Now to point to correct Nectar models. Open up the source and change the associations from favorite products model to Nectar models. In the end we have a schema like:
```

Comment:
We end with saying what the schema would look like and then start with a statement rather that directly taking about the schema.. can we please redo this part.